apps/functions-subscription/src/adapters/azure/functions/list-trials.ts

```diff
@@ -0,0 +1,76 @@
+import * as H from '@pagopa/handler-kit';
+import { pipe, flow } from 'fp-ts/lib/function';
+import * as RTE from 'fp-ts/ReaderTaskEither';
+import * as t from 'io-ts';
+import { httpAzureFunction } from '@pagopa/handler-kit-azure-func';
+import { TrialPaginatedCollection } from '../../../generated/definitions/internal/TrialPaginatedCollection';
+import { SystemEnv } from '../../../system-env';
+import { getAndValidateUser, parseQueryParameter } from './middleware';
+import { toHttpProblemJson } from './errors';
+import { toTrialListAPI } from './codec';
+import { TrialIdCodec } from '../../../domain/trial';
+import {
+  IWithinRangeIntegerTag,
+  NumberFromString,
+  WithinRangeInteger,
+} from '@pagopa/ts-commons/lib/numbers';
+import { withDefault } from '@pagopa/ts-commons/lib/types';
+
+type Env = Pick<SystemEnv, 'listTrials'>;
+
+export type QueryPageSize = t.TypeOf<typeof QueryPageSizeBase>;
+const QueryPageSizeBase = t.union([
+  WithinRangeInteger<1, 100, IWithinRangeIntegerTag<1, 100>>(1, 100),
+  t.literal(100),
+]);
```

Comment:
this type was generated through `yarn generate` starting from the schema for `queryPageSize` and is used to validate the incoming pageSize parameter.
we put an upper limit of 100 to avoid queries of 100+ trials from the api. it defaults to 25.

lmk if we should increase or lower the limit!

Reply:
you are right, i forgot the values inside the OpenAPI. 😄